File: Step 1 - Learn the Basics/1.4 Know Basic Maths/02 Reverse a Number/reverse_integer.py
# https://leetcode.com/problems/reverse-integer/description/

def reverse(self, x):
    neg = x < 0
    string = str(x)[::-1] if not neg else str(x)[:0:-1]
    result = int(string)
    if neg:
        result = -result
    if result < -2**31 or result > 2**31 - 1:
        return 0
    return result

# Test the function
x = 123  # Change input here for testing
print(f"Reversed: {reverse(x)}")

File: Step 1 - Learn the Basics/1.4 Know Basic Maths/02 Reverse a Number/ReverseInteger.java
// https://leetcode.com/problems/reverse-integer/description/

public class ReverseInteger {
    public static int reverse(int x) {
        long ans = 0;

        while (x != 0) {
            ans = ans * 10 + x % 10;
            x /= 10;
        }

        return (ans < Integer.MIN_VALUE || ans > Integer.MAX_VALUE) ? 0 : (int) ans;
    }

    public static void main(String[] args) {
        int x = 123; // Change the input here for testing
        System.out.println("Reversed: " + reverse(x));
    }
}
